src/main/java/org/spongepowered/common/item/inventory/query/Query.java

```diff
@@ -244,6 +257,30 @@ private IntSet getSlots(Collection<Lens<TInventory, TStack>> lenses) {
         return new Query<>(adapter, Type.COMPOUND, args);
     }
 
+    public static <TInventory, TStack> Query<TInventory, TStack> intersect(InventoryAdapter<TInventory, TStack> adapter, Object... args) {
+        return new Query<>(adapter, Type.INTERSECT, args);
+    }
+
+    public static <TInventory, TStack> Query<TInventory, TStack> union(InventoryAdapter<TInventory, TStack> adapter, Object... args) {
+        List<Fabric<TInventory>> fabricList = new ArrayList<>();
```

Comment:
Potentially use builder pattern for `CompoundFabric` & `CompoundLens` rather than list passed to ctor lower down?